### PR TITLE
feat: add gpu dockerfile support to jina hub new

### PR DESCRIPTION
**Pull request by alaeddine-13**
**Goals:**
* Add tf, jax, pytorch and generic gpu dockerfile template to jina hub new command

### CODE REVIEW

File: jina/hubble/hubio.py [hunk 1]
```diff
@@ -117,13 +117,48 @@ def new(self) -> None:
...
+                )
+            )
+
+            is_dockerfile = self.args.dockerfile or Prompt.ask(
+                ':grey_question: Select how you want to generate the [bold]Dockerfile[/bold] for this Executor?',
+                choices=['cpu', 'torch-gpu', 'tf-gpu', 'none'],
+                default='cpu',
+            )
+
+            print(
```
Comment (alaeddine-13) [inline jina/hubble/hubio.py:152]:
```suggestion
            if isinstance(is_dockerfile, string) and 'gpu' in is_dockerfile:
                print((Panel.fit('GPU dockerfiles expect requirements to come from "gpu_requirements.txt", make sure to provide extra dependencies in this file.')
            print(
```

File: jina/resources/executor-template/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,13 @@
-FROM jinaai/jina:latest
+ARG JINA_VERSION=latest
+
+FROM jinaai/jina:${JINA_VERSION}
```
Comment (alaeddine-13) [inline jina/resources/executor-template/Dockerfile:3]: nice

**Review by alaeddine-13 — COMMENTED** [review]
let's also mention those templates and build args in the docs ?
I think users should be aware that they can do things like:
```shell
docker build --build-arg JINA_VERSION=3.8.0 --build-arg TF_PACKAGE_VERSION=2.0.0
```

File: docs/fundamentals/executor/hub/create-hub-executor.md [hunk 3]
```diff
@@ -51,6 +51,11 @@ structure.
     package.
 
 
+```{tip}
+In the wizard of `jina hub new`, you can choose from four Dockerfile templates: `cpu`, `tf-gpu`, `torch-gpu`, and `jax-cpu`.
```
Comment (hanxiao) [inline docs/fundamentals/executor/hub/create-hub-executor.md:55]:
```suggestion
In the wizard of `jina hub new`, you can choose from four Dockerfile templates: `cpu`, `tf-gpu`, `torch-gpu`, and `jax-gpu`.
```

**Pull request merged**